casepro/statistics/signals.py

```diff
@@ -38,6 +41,29 @@ def record_new_outgoing(sender, instance, created, **kwargs):
         if instance.partner:
             DailyCount.record_item(day, DailyCount.TYPE_REPLIES, partner)
 
+        if case:
+            # count the very first response on an org level
+            if instance == case.outgoing_messages.earliest('created_on'):
```

Comment:
There's two workflows we need to support here - first is:

1. Helpdesk admin assigns a case to a partner
2. Partner replies to contact
3. Partner responsiveness is time between being assigned and reply sent

Second is:

1. Partner opens case from messages they have label access to
2. Partner replies to contact (likely immediately)
3. Partner responsiveness is time between being message being received and reply sent

UPartner deployments use both workflows. For the second workflow, only measuring time between assignment and replying is at best meaningless and at worst misleading.

I gather it's not an immediate requirement for you guys, but time between a message being received, and a reply being being sent is meaningful for any deployment. It tells about the quality of service from the point of view of the contact, so I think that's something that should be recorded regardless of what else is recorded.

Where cases are being assigned to partners, then it's also clearly useful for any deployment to know how quickly partner orgs respond. So maybe that's something we record separately as "time between assign/re-assign and first reply from the partner". It would not apply when the partner self-assigns the case.

That make sense?